### DISCUSSION TITLE
Could not build on Apple Silicon M2

### DISCUSSION THREAD

**Discussion by dzungpv** [revk/xESP32-Faikout#467] [General]
I try to build on Apple Silicon M2 but show error: 
```
revk_settings.c:5:10: fatal error: 'popt.h' file not found
```
I try install popt with brew:  

> brew install popt

There is no libpopt-dev in Mac https://github.com/revk/E3D/issues/6

But still the same, what OS you are using to build the code @revk ?

**Comment by revk** [comment]
I build on Linux and Mac.

Popt would be one of the tools used. Not sure you need for just building the ESP target.

Which object is it expecting popt for?

**Reply by dzungpv** [reply]
I enter command make pico to build. And it show error, you seem use popt to generate settings file.

**Reply by revk** [reply]
Ah, that makes sense. I'll have a look in my Mac as I definitely manage to build.

**Reply by revk** [reply]
OK it works for me, so somehow I have managed to install popt

<img width="566" alt="Screenshot 2024-09-04 at 08 00 16" src="https://github.com/user-attachments/assets/2abfa474-aa2d-49d2-9482-5ac65cadecc8">

**Reply by dzungpv** [reply]
I use Mac Sonoma 14.6.1 (23G93), M2 Max. Do you use Intel mac? and how do you install popt?

**Reply by revk** [reply]
It is not intel, no. A Mac Studio.

**Reply by dzungpv** [reply]
![faikin](https://github.com/user-attachments/assets/cf2930d7-3b93-46b9-8971-14ca22ef1572)
Strange, what is your Mac OS version? I use ESP IDF 5.2.2, Macbook Pro, Sonoma 14.6.1.

**Reply by revk** [reply]
Well I vaguely recall having to install popt, and I use brew, but I really don't know which package provided it.

**Reply by dzungpv** [reply]
> Well I vaguely recall having to install popt, and I use brew, but I really don't know which package provided it.

Can you send me output of the command:  brew info popt ?
My version is 1.19, it may cause the problem

```
 brew info popt        
==> popt: stable 1.19 (bottled)
Library like getopt(3) with a number of enhancements
https://github.com/rpm-software-management/popt
Not installed
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/p/popt.rb
License: MIT
==> Analytics
install: 4,295 (30 days), 13,572 (90 days), 46,156 (365 days)
install-on-request: 54 (30 days), 168 (90 days), 827 (365 days)
build-error: 0 (30 days)
```

**Reply by revk** [reply]
<img width="649" alt="Screenshot 2024-09-04 at 08 35 38" src="https://github.com/user-attachments/assets/38b87493-4670-41f7-a19d-a28f8fa9199f">

**Reply by dzungpv** [reply]
> <img alt="Screenshot 2024-09-04 at 08 35 38" width="649" src="https://private-user-images.githubusercontent.com/996983/364271042-38b87493-4670-41f7-a19d-a28f8fa9199f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU0MzU3NDIsIm5iZiI6MTcyNTQzNTQ0MiwicGF0aCI6Ii85OTY5ODMvMzY0MjcxMDQyLTM4Yjg3NDkzLTQ2NzAtNDFmNy1hMTlkLWEyOGY4ZmE5MTk5Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwNFQwNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MmZiY2E3MTZkZWNkMzk4NjU0ZWYxOGNhNzYwYzg1Mzc5Zjk5NzEyMzQ4MjNhMjEzYzIzZDIzMDUyZTlmOTcyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.aEyJclWupbluJjnRHEi68TCJqWVEOpBWwRth7ojZd_s">

I see difference now, under https://github.com/rpm-software-management/popt your show: **Installed** and with me show **Not installed**

**Reply by dzungpv** [reply]
Ah, I have just uninstall it to check now it is same with you but still not working,
![popt2](https://github.com/user-attachments/assets/962d19ca-981b-4f90-b066-fc7746e8c05c)

**Reply by revk** [reply]
Sorry I am really not an expert on `brew`.

**Reply by dzungpv** [reply]
I will try with other machine, it may cause by the latest Mac OS

**Comment by dzungpv** [comment]
I have fixed myself;
change https://github.com/revk/ESP32-Faikin/blob/main/Tools/Simulators/Makefile:
```
INCLUDES := -I/usr/local/include/
LIBS := -L/usr/local/Cellar/popt/1.18/lib/
```
to 

```
INCLUDES := -I/opt/homebrew/include/
LIBS := -L/opt/homebrew/Cellar/popt/1.19/lib/
```

And it build OK. Just add to note because default setting for Intel chip

**Reply by revk** [reply]
Thanks for confirming.
